client/js/components/launchdarkly.js

```diff
@@ -0,0 +1,27 @@
+import LDClient from 'ldclient-js';
+import uuidV1 from 'uuid/v1';
+
+const LDUserKey = 'LDUserKey';
+
+let uuid = window.localStorage.getItem(LDUserKey);
+
+if (!uuid) {
+  uuid = uuidV1();
+  window.localStorage.setItem(LDUserKey, uuid);
+}
+
+const user = {
+  key: uuid,
+  anonymous: true
+};
+
+const ldclient = LDClient.initialize('58e63fe26b7519095c7484cb', user);
+
+const getFlags = new Promise((resolve, reject) => {
+  ldclient.on('ready', () => {
+    const allFlags = ldclient.allFlags();
+    resolve(allFlags);
+  });
```

Comment:
Did you get to the bottom of rejecting on timeout and/or with any errors back from LD?